tests/binary_deserialization.cpp

```diff
@@ -776,4 +776,78 @@ TEST_F(recv, should_convert_TIMESTAMPOID_to_time_point) {
     EXPECT_EQ(result, expected);
 }
 
+struct recv_microseconds_ok : recv, WithParamInterface<std::pair<std::vector<char>, std::chrono::microseconds>> {
+    static_assert(std::chrono::microseconds::min().count() == std::numeric_limits<int64_t>::min());
+    static_assert(std::chrono::microseconds::max().count() == std::numeric_limits<int64_t>::max());
+};
+
+TEST_P(recv_microseconds_ok, should_convert_INTERVALOID_to_chrono_microseconds) {
+    auto [buffer, expected] = GetParam();
+
+    EXPECT_CALL(mock, field_type(_)).WillRepeatedly(Return(1186));
+    EXPECT_CALL(mock, get_value(_, _)).WillRepeatedly(Return(buffer.data()));
+    EXPECT_CALL(mock, get_length(_, _)).WillRepeatedly(Return(16));
+    EXPECT_CALL(mock, get_isnull(_, _)).WillRepeatedly(Return(false));
+
+    std::chrono::microseconds result;
+    ozo::recv(value, oid_map, result);
+
+    EXPECT_EQ(result, expected);
+}
+
+inline auto make_param(std::array<unsigned char, 16> data, int64_t microseconds) {
+    std::vector<char> buffer(16, 0);
+    std::transform(data.cbegin(), data.cend(), buffer.begin(), [](auto c) { return static_cast<char>(c); });
+
+    return std::make_pair(buffer, std::chrono::microseconds(microseconds));
```

Comment:
You have a copy of the `buffer` here. Boost.Ranges library is good for such a thing.

```c++
const auto to_signed_char = boost::adaptors::transformed([](unsigned char c) {
    return static_cast<char>(c); 
});

//...

auto buffer = data | to_signed_char;
return std::make_pair(std::vector<char>{buffer.begin(), buffer.end()}, 
    std::chrono::microseconds(microseconds));
```